backend/src/plugins/Automod/AutomodPlugin.ts

```diff
@@ -134,6 +134,18 @@ const configPreprocessor: ConfigPreprocessorFn<AutomodPluginType> = options => {
       }
 
       if (rule["actions"]) {
+        if (rule["actions"].change_roles && (rule["actions"].add_roles || rule["actions"].remove_roles)) {
+          throw new StrictValidationError([
+            `Can't use both 'change_roles' and 'add_roles'/'remove_roles' at rule '${rule.name}'`,
+          ]);
+        }
+
+        if (rule["actions"].add_roles && rule["actions"].remove_roles) {
+          throw new StrictValidationError([
+            `Can't use both 'add_roles' and 'remove_roles' at rule '${rule.name}', use 'change_roles' instead`,
+          ]);
+        }
+
         for (const actionName in rule["actions"]) {
```

Comment:
And if you want to keep going, feel free to replace all the rule["actions"] in this file with this
```suggestion
        for (const actionName in rule.actions) {
```